### PR TITLE
Start of hardening Zathras

### PR DESCRIPTION
**Pull request by dvalinrh**
Keith is the reviewer, rest of you are fyi.

This code will not be merged until all changes are done, and testing is completed 100%

### CODE REVIEW

File: ansible_roles/roles/terminate_on_error/tasks/main.yml [hunk 1]
```diff
@@ -13,6 +13,13 @@
   no_log: true
   failed_when: "'IGNORE_IT' in status.stderr"
 
+- name: Read status file
+  shell: cat {{ working_dir }}/{{ status_file }} 2> /dev/null
+  register: status1
+  ignore_errors: yes
+  no_log: true
+  failed_when: "'fail' in status.stdout"
+
```
Comment (kdvalin) [inline ansible_roles/roles/terminate_on_error/tasks/main.yml]: Why is this block duplicated from lines 9-14?

File: tools_bin/update_system [hunk 2]
```diff
@@ -180,8 +180,7 @@ iso_retrieve_image()
 		rm dvd1.iso 2> /dev/null
 		wget $1 2> /dev/null
 		if [ $? -ne 0 ]; then
-			echo Unable to wget $1
-			exit 1
+			erorr_out "Unable to wget $1"
```
Comment (kdvalin) [inline tools_bin/update_system]: spelling mistake

File: bin/burden [hunk 3]
```diff
@@ -2507,7 +2566,7 @@ run_scenario()
 				pindex=0
 				pid=""
 			else
-				tmpfile=$(mktemp /tmp/zath_temp.XXXXXX)
+				tmpfile=$(mktemp /tmp/zath_temp4.XXXXXX)
```
Comment (kdvalin) [inline bin/burden]: What's the purpose of `zath_temp[0-9].XXXXX`?

Name does not convey the intent of the file

File: bin/burden [hunk 4]
```diff
@@ -1790,6 +1791,8 @@ create_ansible_options()
 			popd > /dev/null
 		fi
 		if [ $gl_preflight -eq 0 ]; then
+			# DJV
+			#cleanup_and_exit "No execute" 1
```
Comment (kdvalin) [inline bin/burden]: Clean out commented code

File: tools_bin/update_system [hunk 5]
```diff
@@ -180,7 +180,7 @@ iso_retrieve_image()
 		rm dvd1.iso 2> /dev/null
 		wget $1 2> /dev/null
 		if [ $? -ne 0 ]; then
-			erorr_out "Unable to wget $1"
+			erorr_out "Not able to perform wget $1"
```
Comment (kdvalin) [inline tools_bin/update_system]: Should be `error_out`, not `erorr_out`

File: bin/burden (hunk 4 above)
Comment (kdvalin) [reply bin/burden]: Bumping this

Remove commented out code

File: bin/burden [hunk 6]
```diff
@@ -1736,21 +1804,28 @@ create_ansible_options()
...
-			done
-			pindex=0
+		if [ $gl_preflight -eq 0 ]; then
+			# DJV
+			#cleanup_and_exit "No execute" 1
+			if [[ "$gl_ssh_key_file" == "" ]]; then
+				kick_off.sh -t $gl_top_dir -d $run_dir -f $gl_sys_file -a $gl_create_attempts -S $gl_spot_recover -c $gl_test_def_dir $rm_dirs | tee ${run_dir}/ansible_log &
+			else
+				kick_off.sh -t $gl_top_dir -d $run_dir -f $gl_sys_file -a $gl_create_attempts -S $gl_spot_recover -s $gl_ssh_key_file -c $gl_test_def_dir $rm_dirs | tee ${run_dir}/ansible_log &
+			fi
```
Comment (kdvalin) [inline bin/burden]: Consider using the if to build different options between these 2 cases in a variable.

Less copy-paste errors later

File: tools_bin/update_system (hunk 5 above)
Comment (kdvalin) [reply tools_bin/update_system]: Bump

Typo, should be `error_out`

File: ansible_roles/roles/test_generic/tasks/main.yml [hunk 7]
```diff
@@ -79,7 +79,7 @@
       msg: "{{ dest_dir }}/{{ test_data.exec_dir }}/{{ test_data.test_script_to_run }} --run_user {{ config_info.test_user }} --home_parent {{ config_info.user_parent_home_dir }} --iterations {{ config_info.test_iterations }} --tuned_setting {{ sys_confg }} --host_config \"{{ config_info.host_config_info }}\" --sysname \"{{ config_info.host_or_cloud_inst }}\" --sys_type {{ config_info.system_type }} {{ test_data.test_specific }}"
 
   - name: record the shell using
-    shell: "echo \"#/bin/bash\" >> /tmp/{{ test_data.test_name }}.cmd"
+    shell: "echo \"#/bin/bash\" > /tmp/{{ test_data.test_name }}.cmd"
```
Comment (kdvalin) [inline ansible_roles/roles/test_generic/tasks/main.yml:82]: Seems like this was in the tree before #41 was opened, guess I got impatient :)

Comment (kdvalin) [reply ansible_roles/roles/test_generic/tasks/main.yml:82]: Will close that one out

File: bin/burden (hunk 6 above)
Comment (dvalinrh) [reply bin/burden]: Done.

**Review by kdvalin — APPROVED** [review]
LGTM

**Comment by dvalinrh** [comment]
We are merging the first round of changes, else things are going to get a bit messy.

**Pull request merged**